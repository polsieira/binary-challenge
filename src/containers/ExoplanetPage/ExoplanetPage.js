import React from 'react';
import { connect } from 'react-redux';
import './ExoplanetPage.scss';
import PropTypes from 'prop-types';


export const ExoplanetPage = ({ match, exoplanets }) => {
  const exoplanetName = match.match.url.split('/')[2];
  const exoplanet = exoplanets.find(exoplanet => exoplanet.pl_name === exoplanetName);
  const { pl_name, pl_hostname, pl_orbper, pl_bmassj, gaia_dist, pl_disc, pl_pelink } = exoplanet;
  return (
    <div className='ExoplanetPage'>
      <div className='main-info'>
        <a className='planet-name' href={pl_pelink}>{pl_name}</a>
        <p className='discovery'>{` | ${pl_disc}`}</p>
      </div>
      <p className='star-name'>{pl_hostname}</p>
      <div className='data'>
        <p className='label'>{'Orbital Period:'}</p>
        <p className='orbital-period'>{`${pl_orbper} days`}</p>
        <p className='label'>{'Mass:'}</p>
        <p className='mass'>{`Mass: ${pl_bmassj} Mass of Jupiter`}</p>
        <p className='label'>{'Distance To:'}</p>
        <p className='distance'>{`${gaia_dist} light years`}</p>
      </div>
    </div>
  )
}

export const mapStateToProps = ({ exoplanets }) => ({
  exoplanets
})

export default connect(mapStateToProps)(ExoplanetPage);

ExoplanetPage.propTypes = {
  exoplanets: PropTypes.array,
};