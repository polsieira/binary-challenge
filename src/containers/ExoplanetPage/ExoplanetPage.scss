.ExoplanetPage {
  background-image: url('https://www.thelocal.ch/userdata/images/article/9b4ead50aacc44f9099cc1f64ec7e870a33a8fc2625583db0702c53e34fc186d.jpg');
  height: 100vh;
  width: 100%;

  .main-info {
    display: flex;

    .planet-name {
      color: white;
      cursor: pointer;
      font-size: 40px;
      font-weight: 900;
      margin: 0px;
      padding: 40px 0px 0px 40px;
      text-align: left;
      text-decoration: none;
    }
    .discovery {
      color: white;
      font-size: 40px;
      font-weight: 300;
      margin: 0px;
      padding: 40px 0px 0px 10px;
      text-align: left;
    }
  }

  .star-name {
    color: white;
    font-size: 20px;
    font-weight: 500;
    margin: 0px;
    padding: 10px 45px;
    text-align: left;
  }

  .orbital-period,
  .mass,
  .distance, 
  .link {
    color: white;
    text-align: left;
  }
}