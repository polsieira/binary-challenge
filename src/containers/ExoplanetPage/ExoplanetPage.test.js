import React from 'react';
import { shallow } from 'enzyme';
import { ExoplanetPage, mapStateToProps, mapDispatchToProps } from './ExoplanetPage';

describe('ExoplanetPage', () => {
  describe('ExoplanetPage component', () => {
    let wrapper;
    const mockMatch = {
      match: {
        url: 'blah/blah/exoplanet'
      }
    };
    const mockExoplanets = [{
      pl_name: 'exoplanet',
      pl_hostname: 'star',
      pl_orbper: 2000,
      gaia_dist: 100,
      pl_disc: 2005
    }]

    beforeEach(() => {
      wrapper = shallow(<ExoplanetPage match={mockMatch} exoplanets={mockExoplanets} />);
    });

    it('should match the snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });

  });

  describe('mapStateToProps', () => {
    it('should return an object with exoplanets', () => {
      const mockState = {
        exoplanets: [{
          pl_name: 'exoplanet',
          pl_hostname: 'star',
          pl_orbper: 2000,
          gaia_dist: 100,
          pl_disc: 2005
        }],
        error: '',
        isLoading: false
      };

      const expected = {
        exoplanets: [{
          pl_name: 'exoplanet',
          pl_hostname: 'star',
          pl_orbper: 2000,
          gaia_dist: 100,
          pl_disc: 2005
        }]
      };

      const mappedProps = mapStateToProps(mockState);

      expect(mappedProps).toEqual(expected);
    });
  });

});