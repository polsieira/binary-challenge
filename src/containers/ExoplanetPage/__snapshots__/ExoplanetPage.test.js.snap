// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExoplanetPage ExoplanetPage component should match the snapshot 1`] = `
<div
  className="ExoplanetPage"
>
  <div
    className="main-info"
  >
    <a
      className="planet-name"
    >
      exoplanet
    </a>
    <p
      className="discovery"
    >
       | 2005
    </p>
  </div>
  <p
    className="star-name"
  >
    star
  </p>
  <div
    className="data"
  >
    <p
      className="label"
    >
      Orbital Period:
    </p>
    <p
      className="orbital-period"
    >
      2000 days
    </p>
    <p
      className="label"
    >
      Mass:
    </p>
    <p
      className="mass"
    >
      Mass: undefined Mass of Jupiter
    </p>
    <p
      className="label"
    >
      Distance To:
    </p>
    <p
      className="distance"
    >
      100 light years
    </p>
  </div>
</div>
`;
